File: JAVA_COURSE/JAVA/08_Robot11/src/RobotManager11.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author LV
 */
public class RobotManager11 {

    /**
     * @param args the command line arguments
     */
       public static void main(String[] args) {
        // Создаем объект для управления роботом
        RobotManager11 rm = new RobotManager11();
 
        // Создаем объекта класса Robot
        Robot robot = new Robot();
        // Курс 45 градусов
        robot.setCourse(45);
        // Напечатать курс
        System.out.println(robot.getCourse());      // Здесь будет 45
 
        // вызываем метод и передаем туда робота
        rm.changeCourse(robot);
 
        // Напечатать курс
        System.out.println(robot.getCourse());      // Здесь будет 180
    }
 
    private void changeCourse(Robot robot) {
       robot = new Robot();
        robot.setCourse(180);
    }
    }
    

File: JAVA_COURSE/KOTLIN/08_Robot11/src/kotlinrobot/RobotManager11.kt
package kotlinrobot

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author LV
 */
class RobotManager11 {

    private fun changeCourse(robot: Robot) {
        var robot = robot
        robot = Robot()
        robot.course = 180.0
    }

    companion object {

        /**
         * @param args the command line arguments
         */
        @JvmStatic
        fun main(args: Array<String>) {
            // Создаем объект для управления роботом
            val rm = RobotManager11()

            // Создаем объекта класса javarobot.Robot
            val robot = Robot()
            // Курс 45 градусов
            robot.course = 45.0
            // Напечатать курс
            println(robot.course)      // Здесь будет 45

            // вызываем метод и передаем туда робота
            rm.changeCourse(robot)

            // Напечатать курс
            println(robot.course)      // Здесь будет 180
        }
    }
}
    
